src/test/java/ComplicatedOperationServiceTest.java

```diff
@@ -0,0 +1,392 @@
+import calculator.model.Input;
+import calculator.service.BasicOperationService;
+import calculator.service.ComplicatedOperationService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.internal.matchers.Null;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.hamcrest.Matchers.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ComplicatedOperationServiceTest {
+
+    private ComplicatedOperationService cOS;
+    //Inputs for sum, subtraction, multiplication, and division
+    private Input posAPosB;
+    private Input posANegB;
+    private Input negAPosB;
+    private Input negANegB;
+    private Input posAZeroB;
+    private Input zeroAPosB;
+    private Input zeroAZeroB;
+    private Input posANullB;
+    private Input nullAPosB;
+
+    private Input nullInput;
+    private Input nullANullB;
+
+    private List<Double> posPos;
+    private List<Double> posNeg;
+    private List<Double> negPos;
+    private List<Double> negNeg;
+    private List<Double> zeroPos;
+
+
+    @Mock
+    private BasicOperationService bOS;
+
+    @Before
+    public void setup() {
+        //Inputs setup
```

Comment:
Its completely acceptable, but you don't need to set all the values in the @Before function (unless they're really reusable) - sometimes it would be just ok to declare the Input inside of the test - probably more visible for me, but it's up to you
Definitely no need to declare nullInput - you can just put null inside of testing method